srcpkgs/papis/template

```diff
@@ -1,22 +1,34 @@
 # Template file for 'papis'
 pkgname=papis
-version=0.11.1
-revision=2
+version=0.12
+revision=1
 build_style=python3-module
 hostmakedepends="python3-setuptools"
-depends="python3-requests python3-yaml python3-chardet python3-BeautifulSoup4
- python3-colorama python3-click python3-slugify python3-prompt_toolkit
- python3-tqdm python3-Pygments python3-stevedore python3-parsing
- python3-filetype python3-bibtexparser python3-habanero python3-arxiv2bib
- python3-isbnlib python3 python3-setuptools python3-lxml
- python3-typing_extensions python3-doi"
+depends="python3-requests python3-filetype python3-parsing python3-arxiv2bib
+ python3-yaml python3-chardet python3-BeautifulSoup4 python3-colorama
+ python3-bibtexparser python3-click python3-habanero python3-isbnlib
+ python3-prompt_toolkit python3-tqdm python3-Pygments python3-stevedore
+ python3-doi python3-lxml python3-slugify python3-Whoosh"
+checkdepends="${depends} python3-pytest"
 short_desc="Command-line based document and bibliography manager"
 maintainer="xaltsc <xaltsc@protonmail.ch>"
 license="GPL-3.0-or-later"
 homepage="https://github.com/papis/papis"
+changelog="https://github.com/papis/papis/blob/master/CHANGELOG.md"
 distfiles="https://github.com/papis/papis/archive/v${version}.tar.gz"
-checksum=8ebb2f6f50e2e92e16a37ffc1a054983f6b50c6a56d3629d9f612f4197d7a7da
+checksum=75a91f827927a79ac648fbd810aacb1d2a0ba475f69f2f61c421c8b1dbf10cc9
 
-pre_build() {
-	sed -i '/configparser/d' setup.py
+do_patch() {
+	vsed -e '/configparser/d' -i setup.py
+	vsed -e '/--cov=papis/d' -i setup.cfg
+}
+
+do_check() {
+	# Require to generate egg_info so that tests can use entry points
+	python3 setup.py egg_info
+
+	# Deselected one test which fails dring upstream CI too
+	# https://github.com/papis/papis/runs/6558190874?check_suite_focus=true
+	PYTHONPATH="$(cd build/lib* && pwd)" python3 -m pytest papis/ tests/ \
+		--deselect=tests/downloaders/test_project_euclid.py::test_1
```

Comment:
this can be turned into
```sh
make_check_pre="env PYTHONPATH=$wrksrc/build/lib"
make_check_args="--deselect=tests/downloaders/test_project_euclid.py::test_1"
make_check_target="papis tests"
```
and then defining do_check() is no longer needed